lib/driver.js

```diff
@@ -332,6 +324,14 @@ class IosDriver extends BaseDriver {
       return;
     }
 
+      // if we have an ipa file, set it in opts
+      if (this.opts.app) {
```

Comment:
Hmm. `installIpa` also bypasses any reset/no reset logic we have, and always uninstalls. Weird. I think we need to look at the install flow altogether.